font-codegen/src/parsing.rs

```diff
@@ -1398,10 +1401,15 @@ fn get_single_lifetime(input: &syn::PathArguments) -> syn::Result<Option<syn::Li
     match get_single_generic_arg(input)? {
         None => Ok(None),
         Some(syn::GenericArgument::Lifetime(arg)) => Ok(Some(arg.clone())),
-        _ => Err(syn::Error::new(input.span(), "expected single lifetime")),
+        _ => Err(logged_syn_error(input.span(), "expected single lifetime")),
     }
 }
 
+pub(crate) fn logged_syn_error<T: Display>(span: Span, message: T) -> syn::Error {
```

Comment:
So this is an improvement, but as we're only doing this with errors that we create explicitly (as opposed to errors that occur during another parse method, for instance when we do `let name = input.parse::<syn::Ident>()?`).

And unfortunately, errors of the other type (generated in syn, and merely propagated by us) are probably the majority of our errors?

Reply:
When I tried introducing some basic errors, e.g. a type that didn't exist, I did get a trace. In any case, IIUC using our own error that auto-captures traces should entirely resolve this.